From 1261081f4232be122c15790fcd3e33b5d989607d Mon Sep 17 00:00:00 2001
From: Daniel Reis <dgreis@sapo.pt>
Date: Thu, 16 Apr 2015 09:45:39 +0100
Subject: [PATCH] Backport client command discovery optimization

---
 openerp/cli/__init__.py | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/openerp/cli/__init__.py b/openerp/cli/__init__.py
index 31582d9..c061175 100644
--- a/openerp/cli/__init__.py
+++ b/openerp/cli/__init__.py
@@ -56,12 +56,9 @@ def main():
     if len(args) and not args[0].startswith("-"):
         logging.disable(logging.CRITICAL)
         for m in module.get_modules():
-            m = 'openerp.addons.' + m
-            __import__(m)
-            #try:
-            #except Exception, e:
-            #    raise
-            #    print e
+            m_path = module.get_module_path(m)
+            if os.path.isdir(os.path.join(m_path, 'cli')):
+                __import__('openerp.addons.' + m)
         logging.disable(logging.NOTSET)
         command = args[0]
         args = args[1:]
-- 
2.1.4

